src/main/webapp/WEB-INF/view/activity.jsp

```diff
@@ -49,13 +43,14 @@
       %>
         <ul class="mdl-list">
       <%
-        for(Conversation conversation : conversations){
+        for(int i=conversations.size()-1; i>=0; i--){
```

Comment:
Consider using Collections.reverse() in the servlet so the JSP logic doesn't have to make an ordering assumption here. In general JSPs should only contain enough logic to render info. "Business logic" like specific ordering should be in the servlet code where it is easier to test.

Reply:
Good call, I never considered that honestly. I'll keep this in mind for other logic ideas I have and make sure to have them in non JSP files.